build_tools/azure/test_script.sh

```diff
@@ -58,8 +58,10 @@ if [[ "$COVERAGE" == "true" ]]; then
 fi
 
 if [[ "$PYTEST_XDIST_VERSION" != "none" ]]; then
-    XDIST_WORKERS=$(python -c "import joblib; print(joblib.cpu_count(only_physical_cores=True))")
-    TEST_CMD="$TEST_CMD -n$XDIST_WORKERS"
+    XDIST_WORKERS=$(python -c "import joblib; print(joblib.cpu_count())")
```

Comment:
Why is this change needed? Is this caused by a bug or limitation in joblib?


Does this negatively impact other build times?

Reply:
Basically right now, most (all?) of the Linux Azure VMs appears to have 1 physical cores and 2 logical cores [^1]. So we run `pytest -n1` which is a bit weird we may as well not use `pytest-xdist` in this case.

For some reason this does speed-up the Debian build but not the others see more details in https://github.com/scikit-learn/scikit-learn/pull/32031#issuecomment-3236388054.

[^1] this seems weird since those are vCPUs so the number of physical and logical should be the same :shrug:?